src/crush/crush.h

```diff
@@ -315,28 +315,46 @@ struct crush_map {
          */
 	__s32 max_devices;
 
-	/*! choose local retries before re-descent */
+	/*! Backward compatibility tunable. It implements a bad solution
+         * and must always be set to 0 except for backward compatibility
+         * purposes
+         */
 	__u32 choose_local_tries;
-	/*! choose local attempts using a fallback permutation before
-	 *! re-descent */
+	/*! Backward compatibility tunable. It implements a bad solution
+         * and must always be set to 0 except for backward compatibility
+         * purposes
+         */
 	__u32 choose_local_fallback_tries;
-	/*! choose attempts before giving up */
+	/*! Tunable. The default value when the CHOOSE_TRIES or
+         * CHOOSELEAF_TRIES steps are omitted in a rule. See the
+         * documentation for crush_rule_set_step() for more
+         * information
+         */
 	__u32 choose_total_tries;
-	/*! attempt chooseleaf inner descent once for firstn mode; on
-	 *! reject retry outer descent.  Note that this does *not*
-	 *! apply to a collision: in that case we will retry as we used
-	 *! to. */
+        /* obsoleted in firefly */
 	__u32 chooseleaf_descend_once;
```

Comment:
I don't understand what the comment means.  The tunables are all here to preserve legacy behavior, so they don't really become obsolete.  And we set it to true now..